### DISCUSSION TITLE
Using private repositories

### DISCUSSION THREAD

**Discussion by j3k0** [airsdk/apm#28] [Ideas]
Some packages might be distributed in another repository than `airsdk.dev`, I'm thinking for example in companies maintaining private libraries. Having a self-hosted private repository might be handy.

So, what about adding the remote repository URL in the project's `"dependencies"`?

Something like:
```json
        {
            "id": "starling-source",
            "version": "2.7.0",
            "source": "https://repository.airsdk.dev"
        }
```

https://repository.airsdk.dev can be used as a default when `source` isn't specified.

As an alternative, the `apm-repository` could have an "upstream" source. When a package is not found, proxy the same request to another apm-repository (probably always `repository.airsdk.dev`)

**Comment by marchbold** [comment]
Ha, I'd actually added that exact property yesterday to start dealing with locally installed packages and yeah eventually multiple repositories. 

Note, multiple repositories is not a main priority currently I don't want to over complicate the system too early, but I am building it with this in mind.
